edelphi/src/main/java/fi/metatavu/edelphi/pages/panel/admin/report/util/ChartModelProvider.java

```diff
@@ -593,7 +576,45 @@ public static Chart createStackedBarChart(String chartCaption, List<String> cate
     return cwaBar;
   }
 
+  /**
+   * Creates a bubble chart without averages and quartiles
+   * 
+   * @param chartCaption chart caption
+   * @param xLabel x axis label
+   * @param xTickLabels x axis tick labels
+   * @param yLabel y axis label
+   * @param yTickLabels y axis tick labels
+   * @param xAxisLabelRotation x axis label rotation
+   * @param yAxisLabelRotation y axis label rotation
+   * @param values values
+   * @return created chart
+   */
+  @SuppressWarnings ("squid:S00107")
   public static Chart createBubbleChart(String chartCaption, String xLabel, List<String> xTickLabels, String yLabel, List<String> yTickLabels, int xAxisLabelRotation, int yAxisLabelRotation, Double[][] values) {
+    return createBubbleChart(chartCaption, xLabel, xTickLabels, yLabel, yTickLabels, xAxisLabelRotation, yAxisLabelRotation, values, null, null, null, null, null, null);
+  }
+
+  /**
+   * Creates a bubble chart
+   * 
+   * @param chartCaption chart caption
+   * @param xLabel x axis label
+   * @param xTickLabels x axis tick labels
+   * @param yLabel y axis label
+   * @param yTickLabels y axis tick labels
+   * @param xAxisLabelRotation x axis label rotation
+   * @param yAxisLabelRotation y axis label rotation
+   * @param values values
+   * @param avgX average x
+   * @param qX1 x axis 1st quartile
+   * @param qX3 x axis 3rd quartile
+   * @param avgY average y
+   * @param qY1 y axis 1st quartile
+   * @param qY3 y axis 3rd quartile
+   * @return created chart
+   */
+  @SuppressWarnings ("squid:S00107")
+  public static Chart createBubbleChart(String chartCaption, String xLabel, List<String> xTickLabels, String yLabel, List<String> yTickLabels, int xAxisLabelRotation, int yAxisLabelRotation, Double[][] values, Double avgX, Double qX1, Double qX3, Double avgY, Double qY1, Double qY3) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)